### PR TITLE
Rename Endpoint class to Address

### PR DESCRIPTION
**Pull request by kuujo**
Renames the `Endpoint` class to `Address` as it will now be used to represent more than just endpoints.

### CODE REVIEW

**Comment by jplock** [comment]
Could this be replaced with guava’s HostAndPort?

**Comment by johnou** [comment]
@jplock no, guava will be ripped out at some point.

File: agent/src/main/java/io/atomix/agent/AtomixAgent.java [hunk 1]
```diff
@@ -165,7 +165,7 @@ public static void main(String[] args) throws Exception {
     }
   }
 
-  static String[] parseAddress(String address) {
+  static String[] parseInfo(String address) {
     String[] parsed = address.split(":");
     if (parsed.length > 3) {
       throw new IllegalArgumentException("Malformed address " + address);
```
Comment (johnou) [inline agent/src/main/java/io/atomix/agent/AtomixAgent.java]: Trim strings after split?

Comment (kuujo) [reply agent/src/main/java/io/atomix/agent/AtomixAgent.java]: If there's whitespace in the string then it won't be parsed by the command line parser as a single argument and will thus be malformed anyways

**Pull request merged**